programs/util.h

```diff
@@ -37,12 +37,13 @@ extern "C" {
 #include <assert.h>
 #include <sys/types.h>    /* stat, utime */
 #include <sys/stat.h>     /* stat */
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #  include <sys/utime.h>  /* utime */
 #  include <io.h>         /* _chmod */
 #else
+#  include <fcntl.h>      /* AT_FDCWD */
+#  include <sys/stat.h>   /* for utimensat */
```

Comment:
What are the minimum requirements to ensure these files can be included without risk ?

Reply:
So my understanding is that, 
if `PLATFORM_POSIX_VERSION >= 200809L`,
then there is some kind of "guarantee" that these `#include` _must_ be successful ?
A link to a doc source stating such guarantee would be great.

Reply:
No idea. Here's a list of headers that uClibc-ng optionally removes:

```
Makefile.in:HEADERS_RM- := \
Makefile.in:HEADERS_RM- += sgtty.h
Makefile.in:HEADERS_RM-$(HAVE_SHARED)                    += dlfcn.h bits/dlfcn.h
Makefile.in:HEADERS_RM-$(PTHREADS_DEBUG_SUPPORT)         += thread_db.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_ARGP)                += argp.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_BSD_ERR)             += err.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_CRYPT)               += crypt.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_EPOLL)               += sys/epoll.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_FENV)                += fenv.h bits/fenv.h bits/fenvinline.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_FLOATS)              += complex.h fpu_control.h ieee754.h \
Makefile.in:HEADERS_RM-$(findstring y,$(UCLIBC_HAS_FTW)$(UCLIBC_HAS_NFTW))  += ftw.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_FTS)                 += fts.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_GLIBC_CUSTOM_PRINTF) += printf.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_GLOB)                += glob.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_GNU_ERROR)           += error.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_GETOPT_LONG)         += getopt.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_IPV6)                += netinet/ip6.h netinet/icmp6.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_BACKTRACE)           += execinfo.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_LIBICONV)            += iconv.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_LIBINTL)             += libintl.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_LOCALE)              += bits/uClibc_ctype.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_PTY)                 += pty.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_REALTIME)            += mqueue.h bits/mqueue.h sched.h \
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_REGEX)               += regex.h regexp.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_SHADOW)              += shadow.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_SOCKET)              += sys/socket.h bits/socket.h sys/socketvar.h bits/socket_type.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_SYSLOG)              += syslog.h sys/syslog.h bits/syslog*.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_THREADS)             += *thread*.h semaphore.h \
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_THREADS_NATIVE)      += atomic.h bits/atomic.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_UTMP)               += bits/utmp.h utmp.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_UTMPX)               += bits/utmpx.h utmpx.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_WCHAR)               += wchar.h wctype.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_WORDEXP)             += wordexp.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_XATTR)               += sys/xattr.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_XLOCALE)             += xlocale.h
Makefile.in:HEADERS_RM-$(UCLIBC_LINUX_SPECIFIC)          += \
Makefile.in:HEADERS_RM-$(UCLIBC_SUPPORT_AI_ADDRCONFIG)   += ifaddrs.h
Makefile.in:HEADERS_RM-$(UCLIBC_SV4_DEPRECATED)          += ustat.h sys/ustat.h bits/ustat.h
Makefile.in:HEADERS_RM-$(UCLIBC_SUSV3_LEGACY)            += sys/timeb.h regexp.h
Makefile.in:HEADERS_RM-$(UCLIBC_SUSV4_LEGACY)            += utime.h ucontext.h
Makefile.in:HEADERS_RM-$(UCLIBC_HAS_ADVANCED_REALTIME)   += spawn.h
```